script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -138,14 +147,20 @@ class PublishPluginCommand extends PluginCommand {
       _print('$packagesPath is not a valid Git repository.');
       throw ToolExit(1);
     }
-    final GitDir baseGitDir =
+    final GitDir baseGitDir = gitDir ??
         await GitDir.fromExisting(packagesPath, allowSubdirectory: true);
 
     final bool shouldPushTag = getBoolArg(_pushTagsOption);
-    final String remote = getStringArg(_remoteOption);
-    String remoteUrl;
+    _RemoteInfo? remote;
     if (shouldPushTag) {
-      remoteUrl = await _verifyRemote(remote);
+      final String remoteName = getStringArg(_remoteOption);
+      final String? remoteUrl = await _verifyRemote(remoteName);
+      if (remoteUrl == null) {
+        printError(
+            'Unable to find URL for remote $remoteName; cannot push tags');
+        throw ToolExit(1);
+      }
```

Comment:
nits: Can refactor this into `_RemoteInfo()`

Reply:
I think putting very non-trivial logic (running an external process, potentially killing the whole execution) in the constructor of a data class would be much less clear; keeping it as a simple data object with the logic outside it makes the object, and the failure cases, easier to understand IMO.